jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -414,18 +490,20 @@ def _plot_mask(self):
 
         # Loop through all active data in the viewer
         for index, layer_state in enumerate(self.state.layers):
-            comps = [str(component) for component in layer_state.layer.components]
+            lyr = layer_state.layer
+            comps = [str(component) for component in lyr.components]
 
             # Skip subsets
-            if hasattr(layer_state.layer, "subset_state"):
+            if hasattr(lyr, "subset_state"):
                 continue
 
             # Ignore data that does not have a mask component
             if "mask" in comps:
-                mask = np.array(layer_state.layer['mask'].data)
+                mask = np.array(lyr['mask'].data)
 
-                data_x = layer_state.layer.data.get_object().spectral_axis
-                data_y = layer_state.layer.data.get_object().flux.value
+                data_obj = lyr.data.get_object()
```

Comment:
Should we be explicit that `cls=Spectrum1D` here, or is that guaranteed elsewhere? When we support translation to/from `NDDataArray` from https://github.com/astropy/astropy/pull/14175, we should be prepared to be specific.

Reply:
Here, I merely cached the `get_object()` output. The actual call predates this PR, so addressing this is out of scope here.